smslibrary/src/test/java/com/eis/smslibrary/random/RandomSMSPeerGenerator.java

```diff
@@ -0,0 +1,114 @@
+package com.eis.smslibrary.random;
+
+import androidx.annotation.NonNull;
+
+import com.eis.smslibrary.SMSPeer;
+import com.google.i18n.phonenumbers.PhoneNumberUtil;
+import com.google.i18n.phonenumbers.Phonenumber;
+
+import java.util.Random;
+
+/**
+ * Utility class to generate random instances of {@code SMSPeer}. Useful in testing.
+ *
+ * @author Riccardo De Zen
+ */
+public class RandomSMSPeerGenerator implements RandomPeerGenerator<String, SMSPeer> {
+
+    private static final String REGION_ERR = "This region's country code is not supported: ";
+    private static final String INVALID_ADDRESS_HEADER = "#";
+
+    private static final Random random = new Random();
+    private static final PhoneNumberUtil utils = PhoneNumberUtil.getInstance();
+
+    /**
+     * @return a valid built {@code SMSPeer} for the default region.
+     */
+    @NonNull
+    @Override
+    public SMSPeer generateValidPeer() {
+        return new SMSPeer(generateValidAddress());
+    }
+
+    /**
+     * @param region the region for the {@code SMSPeer}.
+     * @return a valid built SMSPeer for the given region.
+     */
+    @NonNull
+    public SMSPeer generateValidPeer(String region) {
+        return new SMSPeer(generateValidAddress(region));
+    }
+
+    /**
+     * @return an example of valid address for the default region.
+     */
+    @NonNull
+    @Override
+    public String generateValidAddress() {
+        return generateValidAddress(SMSPeer.getDefaultRegion());
+    }
+
+    /**
+     * @param region the region for the {@code SMSPeer} address.
+     * @return an example of valid address for the given region.
+     * @throws IllegalArgumentException If the region is not supported.
+     */
+    @NonNull
+    public String generateValidAddress(String region) throws IllegalArgumentException {
+        if (!utils.getSupportedRegions().contains(region))
+            throw new IllegalArgumentException(REGION_ERR + region);
+        String address;
+        SMSPeer peer;
+        do {
+            Phonenumber.PhoneNumber number = getRandomPhoneNumber(region);
+            address = utils.format(number, PhoneNumberUtil.PhoneNumberFormat.E164);
+            peer = new SMSPeer(address);
+        } while (peer.getInvalidityReason() != null);
+        return address;
+    }
+
+    /**
+     * Method to generate an invalid address. As a simple invalidity criteria, a partially random
+     * String is returned.
+     *
+     * @return an invalid address for an {@code SMSPeer}.
+     * @see RandomSMSPeerGenerator#getInvalidRandomString() for details on the Strings returned.
+     */
+    @NonNull
+    @Override
+    public String generateInvalidAddress() {
+        return getInvalidRandomString();
```

Comment:
You might as well move the body of `getInvalidRandomString()` in here since it's the only place where it's used.

Reply:
I dunno, I feel like leaving it separated in case I need to modify the class later or do something in the address method that doesn't involve the String directly, feels cleaner this way to me